### PR TITLE
Convert San d'Oria Missions through 2-2 to Interaction Framework

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Converts San d'Oria Missions to the Interaction Framework.  This slightly depends on the getMissionMask refactor, and is also shown in this PR, as it really makes life easier to test against.  For each converted mission, both getMissionOffset and finishMissionTimeline globals become just a bit shorter, in hopes to finally deprecate those functions.

This also implements resetting missionStatus to 0 on mission complete.

To avoid becoming too unwieldy of a PR, this will be limited to Rank 1 and through 2-2 missions only.  Future conversions to be issued separately.

**Mission Progress:**
Rank 1:
- [x] 1-1 Smash the Orcish Scouts
- [x] 1-2 Bat Hunt
- [x] 1-3 Save the Children

Rank 2:
- [x] 2-1 The Rescue Drill
- [x] 2-2 The Davoi Report

### CODE REVIEW

**Comment by TeoTwawki** [comment]
> in hopes to finally ~~deprecate~~ _**smite**_ those functions _and throw a party at their funerals_.

File: scripts/zones/Southern_San_dOria/npcs/Endracion.lua [hunk 1]
```diff
@@ -85,13 +72,11 @@ entity.onTrigger = function(player, npc)
...
 
 entity.onEventUpdate = function(player, csid, option)
-    -- printf("onUpdateCSID: %u", csid)
-    -- printf("onUpdateOPTION: %u", option)
 end
 
 entity.onEventFinish = function(player, csid, option)
-    -- printf("onFinishCSID: %u", csid)
-    -- printf("onFinishOPTION: %u", option)
+    printf("onFinishCSID: %u", csid)
```
Comment (TeoTwawki) [inline scripts/zones/Southern_San_dOria/npcs/Endracion.lua]: you left these uncommented

Comment (claywar) [reply scripts/zones/Southern_San_dOria/npcs/Endracion.lua]: Bah! And I promised to change that

**Review by TeoTwawki — APPROVED** [review]
looks alright.

we'll deal with the NPC naming later, but soonish. the qm# convention is on its way out as well.

**Comment by claywar** [comment]
ex1.lua!  This is probably the first time (if not only) that I've seen something in game called '!'

**Comment by TeoTwawki** [comment]
> ex1.lua! This is probably the first time (if not only) that I've seen something in game called '!'

I've seen a few in retail that are oddly named (retail has plenty that aren't even valid filenames, including all the ones we called "qm" because `???,lua` isn't valid) but we've had a very long tradition of no punctuation allowed so I am surprised `!.lua` slipped by us all. Recently we've begun just giving human readable "wtf does it DO" names to even the qm NPCs

**Pull request merged**